fetcher/fedora.go

```diff
@@ -0,0 +1,355 @@
+			if ok {
+				var pkgs []Package
+				for _, rpm := range yml.Data.Artifacts.Rpms {
+					pkg, err := rpm.NewPackageFromRpm()
+					if err != nil {
+						return nil, xerrors.Errorf("Failed to build package info from rpm name, err: %w", err)
+					}
+					pkgs = append(pkgs, pkg)
+				}
+				results[version].UpdateList[i].Packages = pkgs
+				results[version].UpdateList[i].ModularityLabel = yml.ConvertToModularityLabel()
+			}
+		}
+	}
+	return results, nil
+}
+
+func fetchFeedFilesFedora(reqs []fetchRequest) ([]FetchResult, error) {
+	if len(reqs) == 0 {
+		return nil, xerrors.New("There are no versions to fetch")
+	}
+	results, err := fetchFeedFiles(reqs)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch. err: %w", err)
+	}
+	return results, nil
+}
+
+func fetchUpdateInfosFedora(results []FetchResult) ([]FetchResult, error) {
+	log15.Info("start fetch updateinfo in repomd.xml")
+	updateInfoReqs, err := extractInfoFromRepoMd(results, "updateinfo", mimeTypeXz)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to extract updateinfo from xml, err: %w", err)
+	}
+
+	if len(updateInfoReqs) == 0 {
+		return nil, fmt.Errorf("No updateinfo field in the repomd")
+	}
+
+	results, err = fetchFeedFiles(updateInfoReqs)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch. err: %w", err)
+	}
+	return results, nil
+}
+
+// variousFlawsPattern is regexp to detect title that omit the part of CVE-IDs by finding both `...` and `various flaws`
+var variousFlawsPattern = regexp.MustCompile(`.*\.\.\..*various flaws.*`)
+
+func parseFetchResultsFedora(results []FetchResult) (FedoraUpdatesPerVersion, error) {
+	updateInfos := make(FedoraUpdatesPerVersion, len(results))
+	for _, r := range results {
+		var updateInfo FedoraUpdates
+		if err := xml.NewDecoder(bytes.NewReader(r.Body)).Decode(&updateInfo); err != nil {
+			return nil, xerrors.Errorf("Failed to decode XML, err: %w", err)
+		}
+		var securityUpdate []FedoraUpdateInfo
+		for _, update := range updateInfo.UpdateList {
+			if update.Type == "security" {
+				cveIDs := []string{}
+				for _, ref := range update.References {
+					var ids []string
+					var err error
+					if isFedoraUpdateInfoTitleReliable(ref.Title) {
+						ids = util.CveIDPattern.FindAllString(ref.Title, -1)
+						if ids == nil {
+							// tyr to correct CVE-ID from description, if title has no CVE-ID
+							// NOTE: If this implementation causes the result of collecting a lot of incorrect information, fix to remove it
+							ids = util.CveIDPattern.FindAllString(update.Description, -1)
+						}
+					} else {
+						ids, err = fetchCveIDsFromBugzilla(ref.ID)
+						if err != nil {
+							return nil, xerrors.Errorf("Failed to fetch CVE-IDs from bugzilla, err: %w", err)
+						}
+					}
+					if ids != nil {
+						cveIDs = append(cveIDs, ids...)
+					}
+				}
+				update.CVEIDs = util.UniqueStrings(cveIDs)
+				securityUpdate = append(securityUpdate, update)
+			}
```

Comment:
How about rejecting all but `security` early on like this?
```suggestion
			if update.Type != "security" {
				continue
			}
			cveIDs := []string{}
			for _, ref := range update.References {
				var ids []string
				var err error
				if isFedoraUpdateInfoTitleReliable(ref.Title) {
					ids = util.CveIDPattern.FindAllString(ref.Title, -1)
					if ids == nil {
						// tyr to correct CVE-ID from description, if title has no CVE-ID
						// NOTE: If this implementation causes the result of collecting a lot of incorrect information, fix to remove it
						ids = util.CveIDPattern.FindAllString(update.Description, -1)
					}
				} else {
					ids, err = fetchCveIDsFromBugzilla(ref.ID)
					if err != nil {
						return nil, xerrors.Errorf("Failed to fetch CVE-IDs from bugzilla, err: %w", err)
					}
				}
				if ids != nil {
					cveIDs = append(cveIDs, ids...)
				}
			}
			update.CVEIDs = util.UniqueStrings(cveIDs)
			securityUpdate = append(securityUpdate, update)
```